apis/v1beta1/vspherefailuredomain_types.go

```diff
@@ -88,6 +94,60 @@ type FailureDomainHosts struct {
 	HostGroupName string `json:"hostGroupName"`
 }
 
+// FailureDomainNetwork defines a network configuration that should be used when consuming
+// this failure domain.
+// @rkatz - To be discussed with team, should we just embed NetworkDeviceSpec?
+type FailureDomainNetwork struct {
```

Comment:
Another point:

We may need to set unique identifiers to this to make it work with the CAPI contract for allowing it to be used with ClusterClass.

More information is here: https://main.cluster-api.sigs.k8s.io/developer/providers/migrations/v1.1-to-v1.2.html?highlight=listMapKey#required-api-changes-for-providers

Reply:
yeap, CAPV shouldn't be writing any information back to this structure. This structure should be configured just one time (in fact, the whole failure domain is immutable after created) so I think this is not needed, right?